crates/router/src/core/errors/api_error_response.rs

```diff
@@ -254,3 +255,168 @@ impl actix_web::ResponseError for ApiErrorResponse {
             .body(self.to_string())
     }
 }
+
+impl common_utils::errors::ErrorSwitch<api_models::errors::types::ApiErrorResponse>
+    for ApiErrorResponse
+{
+    fn switch(&self) -> api_models::errors::types::ApiErrorResponse {
+        use api_models::errors::types::{ApiError, ApiErrorResponse as AER};
+
+        let error_message = self.error_message();
+        let error_codes = self.error_code();
+        let error_type = self.error_type();
+
+        match self {
+            Self::NotImplemented { message } => {
+                AER::NotImplemented(ApiError::new("IR", 0, format!("{message:?}"), None))
+            }
+            Self::Unauthorized => AER::Unauthorized(ApiError::new(
+                "IR",
+                1,
+                "API key not provided or invalid API key used", None
+            )),
+            Self::InvalidRequestUrl => {
+                AER::NotFound(ApiError::new("IR", 2, "Unrecognized request URL", None))
+            }
+            Self::InvalidHttpMethod => AER::MethodNotAllowed(ApiError::new(
+                "IR",
+                3,
+                "The HTTP method is not applicable for this API", None
+            )),
+            Self::MissingRequiredField { field_name } => AER::BadRequest(
+                ApiError::new("IR", 4, format!("Missing required param: {field_name}"), None),
+            ),
+            Self::InvalidDataFormat {
+                field_name,
+                expected_format,
+            } => AER::Unprocessable(ApiError::new(
+                "IR",
+                5,
+                format!(
+                    "{field_name} contains invalid data. Expected format is {expected_format}"
+                ), None
+            )),
+            Self::InvalidRequestData { message } => {
+                AER::Unprocessable(ApiError::new("IR", 6, message.to_string(), None))
+            }
+            Self::InvalidDataValue { field_name } => AER::BadRequest(ApiError::new(
+                "IR",
+                7,
+                format!("Invalid value provided: {field_name}"), None
+            )),
+            Self::ClientSecretNotGiven => AER::NotFound(ApiError::new(
+                "IR",
+                8,
+                "Client secret was not provided", None
+            )),
```

Comment:
```suggestion
            Self::ClientSecretNotGiven => AER::BadRequest(ApiError::new(
                "IR",
                8,
                "Client secret was not provided", None
            )),
```

From our existing error mapping.